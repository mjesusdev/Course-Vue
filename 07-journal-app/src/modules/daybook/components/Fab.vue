<template>
    <button class="btn btn-primary">
        <i class="fa fa-2x fa-plus"></i>
    </button>
</template>

<script>
    export default {

    }
</script>

<style lang="scss" scoped>
    button {
        border-radius: 100%;
        bottom: 20px;
        height: 80px;
        position: fixed;
        right: 20px;
        width: 80px;
    }
</style>