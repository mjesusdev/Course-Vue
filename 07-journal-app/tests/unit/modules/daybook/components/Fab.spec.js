import Fab from '@/modules/daybook/components/Fab'
import { shallowMount } from '@vue/test-utils'

describe('Test in FAB component', () => {

    test('should show the icon by default', () => {
        //* fa-plus
        const wrapper = shallowMount( Fab )
        const iTag = wrapper.find('i')

        expect( wrapper.html() ).toMatchSnapshot()
        expect( iTag.classes('fa-plus') ).toBeTruthy()
    })

    test('should show the icon when clicked', () => {
        //* fa-circle
        const wrapper = shallowMount( Fab, {
            props: {
                icon: 'fa-circle'
            }
        })

        const iTag = wrapper.find('i')
        expect( iTag.classes('fa-circle') ).toBeTruthy()
    })

    test('should emit the event "on:click" when do click', () => {
        //* wrapper.emitted('on:click')
        const wrapper = shallowMount( Fab )

        wrapper.find('button').trigger('click')
        expect( wrapper.emitted('on:click') ).toHaveLength(1)
    });

});