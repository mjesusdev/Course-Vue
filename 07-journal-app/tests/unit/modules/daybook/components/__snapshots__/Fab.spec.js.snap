// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FAB component should show the icon by default 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
